### DISCUSSION TITLE
Middle click desktop for window list?

### DISCUSSION THREAD

**Discussion by winteriscariot** [labwc/labwc#2084] [Ideas]
In Openbox you can middle click on the desktop and get a menu of open windows you can select from. Is this planned for labwc? Middle click currently just by default pulls up the normal desktop menu

**Comment by johanmalm** [comment]
It depends on where we get with
https://github.com/labwc/labwc/pull/2035/commits

Suggest we evaluate after that

**Comment by droc12345** [comment]
Are you talking about client-list-{combined}-menu? If so, it's in the works. 
Though if implemented you would have to add the keybind -- Personally I set it to W-enter on my systemn (convenient for me)

**Comment by droc12345** [comment]
I have started to carry patches on my own, they work against the latest labwc git

**Comment by droc12345** [comment]
Client list combined menu has been merged

**Reply by davidphilipbarr** [reply]
I saw, amazing work! Is there a way to have run as a submenu, i see in your screenshots it can but i must be missing something?

**Comment by droc12345** [comment]
WIthout some of my mods, you need to give it a label.

```
<menu id="root-menu" label="">
...
 <menu id="client-list-combined-menu" label="Running..." />
```

I have it as an entry from root-menu.

and a keybind 
```
 <keybind key="A-Space"><action name="ShowMenu" menu="client-list-combined-menu" x.position="center" y.position="center" /></keybind>
```

That puts it in the center of the screen.


Edit to add: I need to undo one of the modifications for it to work properly as a submenu,
It doesn't update properly otherwise.

ETA2: I have a fix submitted, when it gets merged then it will work as submenu as well as keybind.

**Comment by davidphilipbarr** [comment]
Ah yes that seems to be working, i thought i had tried that before, but perhaps not, thanks!

I've just hacked in a few more states - but i get the idea this is a bad way to do it or is it, i'm not that familiar with c?

```
if (view->workspace == workspace) {
				if (view == server->active_view) {
					buf_add(&buffer, "[*] ");
				}
				else if (view->minimized) {
					buf_add(&buffer, "[-] ");
				}
				else if (view->fullscreen) {
					buf_add(&buffer, "[^] ");
				}
				else if (view->maximized) {
					buf_add(&buffer, "[<>] ");
				}
```

**Reply by droc12345** [reply]
Off the top of my head, that looks like it should work.
You might add an ending else that puts a blank in place (so they line up) but up to you.

I'd also probably use a single symbol "^, v, -, *, "  or simply use single letter "m, M, F, A, "

Edit to add: If you are know how to use patches, then  https://patch-diff.githubusercontent.com/raw/labwc/labwc/pull/2153.patch will fix the problem of client list combined not updating as a submenu.

**Reply by davidphilipbarr** [reply]
I've applied the patch and it's now working as expected, thanks!

**Reply by droc12345** [reply]
Fix has been merged, so you won't need the above patch.
